qiskit_experiments/framework/base_experiment.py

```diff
@@ -167,29 +144,114 @@ def _initialize_experiment_data(
 
         return experiment_data._copy_metadata()
 
-    def run_analysis(self, experiment_data, **options) -> ExperimentData:
+    def _pre_transpile_hook(self, backend: Backend):
```

Comment:
I've checked the term "hooking" on the web. I think its meaning is somewhat different from what we do here. You can simply remove the suffix "_hook" from the names of the new methods.

Reply:
Thanks Yeal and Eli for investigation. In the original code we used to add callbacks from library but now logic is hard-coded in the methods. I like the `*_action` and fine with make them public (with a bit of consideration for API stability, i.e. we cannot easily update arg names once we make it public). Please check the update in https://github.com/Qiskit/qiskit-experiments/pull/380/commits/88f1304f7e8fead332f21204551ec056aa8c8ba2